src/lib/normalizer.cc

```diff
@@ -212,13 +220,321 @@ bool Normalizer::VerbalizeSemioticClass(const Token &markup,
+          } else if (token->type() == Token::SEMIOTIC_CLASS) {
+            if (!token->skip()) {
+              LoggerDebug("Verbalizing: [%s]\n", token_form.c_str());
+              string words;
+              if (VerbalizeSemioticClass(*token, &output)) {
+              } else {
+                break;
+                LoggerWarn("First-pass verbalization FAILED for [%s]",
+                           token_form.c_str());
+                // Back off to verbatim reading
+                string original_token = token->name();
+                token->Clear();
+                token->set_name(original_token);
+                token->set_verbatim(original_token);
+                if (VerbalizeSemioticClass(*token, &output)) {
+                  LoggerWarn("Reversion to verbatim succeeded for [%s]",
+                             original_token.c_str());
+                } else {
+                  // If we've done our checks right, we should never get here
+                  LoggerError("Verbalization FAILED for [%s]", token_form.c_str());
+              //continue;
+                break;
+                }
+              }
+            }
+          } else if (token->type() == Token::WORD) {
+            if (token->has_wordid()) {
+                /* Writes original string */
+                if (!compiler(token->wordid(), &output)) {
+                    printf("Failed to compile input string \"%s\"", token->wordid().c_str());
+                    //continue;
+                    break;
+                }
+            } else {
+              LoggerError("Token [%s] has type WORD but there is no word id",
+                          token_form.c_str());
+              //continue;
+              break;
+            }
+          } else {
+            LoggerError("No type found for [%s]", token_form.c_str());
+            //continue;
+            break;
+          }
+ 
+         /* Project and Optimize */       
+         MutableTransducer temp;
+         fst::Project(&output, fst::PROJECT_OUTPUT);
+
+         fst::ShortestPath(output, &temp);
+         output = std::move(temp);
+
+         fst::Minimize(&output);
+
+         fst::Determinize<fst::StdArc>(output, &temp);
+         output = std::move(temp);
+
+        /* Create space fst for concatenation of words */
+         MutableTransducer space;
+         space.AddState();
+         space.AddState();
+         space.AddArc(0, fst::StdArc(0, ' ', 0.0, 1));
+         space.SetStart(0);
+         space.SetFinal(1, 0.0);
+      
+         if (i > 0)
+            fst::Concat(&concatenated_output, space);
+
+         fst::Concat(&concatenated_output, output);
+      }
+      verbalized.push_back(concatenated_output);
+      LoggerDebug("Verbalize output: Words\n%s\n\n", LinearizeWords(&utt).c_str());
+    }
+
+    std::vector<string> verbalized_strings;
+    for (MutableTransducer verb : verbalized) {
+          MutableTransducer shortest_path;
+          fst::ShortestPath(verb, &shortest_path);
```

Comment:
I don't think we want to do this here, if we have cases where the Verbalizer can return multiple paths for one tokenize instance (e.g. for different orderings).